src/appleseed/renderer/modeling/edf/coneedf.cpp

```diff
@@ -195,6 +196,30 @@ namespace
             return sample_cone_uniform_pdf(m_cos_half_angle);
         }
 
+        virtual float get_uncached_max_radiance_value() const APPLESEED_OVERRIDE
+        {
+            Source* radiance_source = m_inputs.source("radiance");
+
+            // If source is not available, simply return max float so that 
+            // light is not considered as a low contributing one.
+            if (!radiance_source)
```

Comment:
This is still not quite correct.

There may well be a source bound to the `radiance` input, but it may not be a uniform source. For instance, it could be a `TextureSource`.

What you need to do is check (1) if there is a source, and (2) if that source is uniform with `Source::is_uniform()`. If both conditions are met, you can use `evaluate_uniform()`. Otherwise, you should return `numeric_limits<float>::max()`.

You'll need to do that for both the `radiance` and `radiance_multiplier` inputs. Only when both have uniform sources bound to them can you return a sensible max radiance value.

Since this code is the same for a few light models, you could factorize it into a helper method in the `EDF` class. This is something that we sometimes do to reduce code duplication. See `ConnectableEntity::check_non_zero_emission()` for an example of how that would look.